platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/accessors/AccessorFragments.kt

```diff
@@ -1037,6 +1110,7 @@ private
 fun TypeAccessibility.Accessible.internalName() =
     type.value.concreteClass.internalName
 
+@Suppress("ERROR_SUPPRESSION")
```

Comment:
❓ Is this really needed? Why?

I tried removing it, things compile and tests pass.
Same question for the next function.

Reply:
Ouch, it's a leftover from an early experiment. `@RequiresOptIn` is a special annotation in that the Kotlin compiler does not allow using it in a type position, so I was trying to workaround that by suppressing the error. The approach I was trying did not work, and then I failed to remove the suppression. I'll fix it.

Reply:
Done in [`94b4038` (#33746)](https://github.com/gradle/gradle/pull/33746/commits/94b4038c04cf5524cf1b3a9702cbfe900d0f9526)